lib/hanami/assets/bundler/compressor.rb

```diff
@@ -0,0 +1,61 @@
+module Hanami
+  module Assets
+    class Bundler
+      # Compresses a JS or CSS file
+      #
+      # @since 0.3.0-add-options-to-javascript-helper
+      # @api private
+      class Compressor
+        # @since 0.3.0-add-options-to-javascript-helper
+        # @api private
+        JAVASCRIPT_EXT      = '.js'.freeze
+
+        # @since 0.3.0-add-options-to-javascript-helper
+        # @api private
+        STYLESHEET_EXT      = '.css'.freeze
+
+        # Return a new instance
+        #
+        # @since 0.3.0-add-options-to-javascript-helper
+        # @api private
+        def initialize(path, configuration)
+          @path = path
+          @configuration = configuration
+        end
+
+        # @return [String, nil] the compressed contents of the file OR nil if it's not compressable
+        #
+        # @since 0.3.0-add-options-to-javascript-helper
+        # @api private
+        def compress
+          case File.extname(@path)
+          when JAVASCRIPT_EXT then _compress(compressor(:js))
+          when STYLESHEET_EXT then _compress(compressor(:css))
+          end
+        end
+
+        private
+
+        # @since 0.3.0-add-options-to-javascript-helper
+        # @api private
+        def compressor(type)
+          @configuration.__send__(:"#{ type }_compressor")
+        end
+
+        # @since 0.3.0-add-options-to-javascript-helper
+        # @api private
+        def _compress(compressor)
+          compressor.compress(@path)
```

Comment:
When this method was in `Bundler`, it wrote the new file too. Instead we just return the string, and let `Bundler` deal with writing (mostly so we don't have to repeat the code for writing/permissions here)